lib/schedule/data.ex

```diff
@@ -322,39 +329,133 @@ defmodule Schedule.Data do
+      blocks: blocks,
+      calendar: gtfs_data.all_modes.calendar,
+      runs: runs,
+      swings: swings
+    }
+  end
+
+  # Parse GTFS files. Returns files parsed without filtering under the key `all_modes`.
+  # Data filtered to only include bus is under the key `bus_only`. Data may appear under one or both
+  # keys depending on how it needs to be used.
+  @spec parse_gtfs_files(gtfs_files()) :: %{bus_only: map(), all_modes: map()}
+  defp parse_gtfs_files(gtfs_files) do
     gtfs_files["feed_info.txt"]
     |> FeedInfo.parse()
     |> FeedInfo.log_gtfs_version()
 
     directions_by_route_id = directions_by_route_id(gtfs_files["directions.txt"])
 
-    bus_routes =
+    all_routes =
       Csv.parse(
         gtfs_files["routes.txt"],
-        filter: &Route.bus_route_valid_row?/1,
+        filter: &Route.row_has_route_type?/1,
         parse: &Route.from_csv_row(&1, directions_by_route_id)
       )
 
-    bus_route_ids = bus_route_ids(bus_routes)
+    all_route_patterns =
+      Csv.parse(gtfs_files["route_patterns.txt"], parse: &RoutePattern.from_csv_row/1)
 
-    gtfs_trips = Gtfs.Trip.parse(gtfs_files["trips.txt"], bus_route_ids)
-    gtfs_trip_ids = MapSet.new(gtfs_trips, & &1.id)
+    all_stop_times_by_trip_id = StopTime.parse(gtfs_files["stop_times.txt"])
 
-    hastus_activities = Hastus.Activity.parse(hastus_files["activities.csv"])
-
-    hastus_trips =
-      Hastus.Trip.parse(hastus_files["trips.csv"])
-      |> Hastus.Trip.expand_through_routed_trips(gtfs_trip_ids)
+    bus_routes = Enum.filter(all_routes, &Route.bus_route_mbta?/1)
+    bus_route_ids = bus_route_ids(bus_routes)
+    bus_trips = Gtfs.Trip.parse(gtfs_files["trips.txt"], bus_route_ids)
+    bus_trip_ids = MapSet.new(bus_trips, & &1.id)
 
-    route_patterns = bus_route_patterns(gtfs_files["route_patterns.txt"], bus_route_ids)
+    bus_stop_times_by_trip_id =
+      Map.filter(all_stop_times_by_trip_id, fn {trip_id, _stop_times} ->
+        MapSet.member?(bus_trip_ids, trip_id)
+      end)
 
-    timepoints_by_id = all_timepoints_by_id(gtfs_files["checkpoints.txt"])
-    timepoint_names_by_id = timepoint_names_for_ids(timepoints_by_id)
+    bus_shapes = shapes_by_route_id(gtfs_files["shapes.txt"], bus_trips)
+
+    %{
+      bus_only: %{
+        routes: bus_routes,
+        route_ids: bus_route_ids,
+        route_patterns: filter_by_route_id(all_route_patterns, bus_route_ids),
+        shapes: bus_shapes,
+        trips: bus_trips,
+        trip_ids: bus_trip_ids,
+        stop_times_by_trip_id: bus_stop_times_by_trip_id
+      },
+      all_modes: %{
+        calendar:
+          Calendar.from_files(gtfs_files["calendar.txt"], gtfs_files["calendar_dates.txt"]),
+        routes: all_routes,
+        route_patterns: all_route_patterns,
+        stops_by_id: all_stops_by_id(gtfs_files["stops.txt"]),
+        stop_times_by_trip_id: all_stop_times_by_trip_id,
+        timepoints_by_id: all_timepoints_by_id(gtfs_files["checkpoints.txt"])
+      }
+    }
```

Comment:
**question:** did you also consider taking an approach of splitting the data up into `bus_only` and `other_modes` to avoid duplication of the bus data, then merging them as-needed? My one worry here is that this might consume a lot of memory.

Reply:
It's probably worth deploying this to a dev environment and seeing how it behaves, especially whether it increases the peak memory usage during startup and schedule processing.

Reply:
Ooh that is an interesting thought! I'll try a deploy and see how the memory looks. If we end up going that route, we still might want to keep a notion of `all_modes` (or maybe `general` or something not mode associated) for the data we currently never filter by mode, like `stops_by_id`, `timepoints_by_id` and `calendar`

Reply:
Looking at this deploy, it did go over the "healthy" threshold for memory -see logs [here](https://mbta.splunkcloud.com/en-US/app/search/search?sid=1671560571.5574339). However, it looks like that has been the trend lately on prod - wondering if it has more to do with the rating switchover + bigger files?

I did a re-deploy of the existing dev-green branch and found that it also was above the "healthy" threshold on startup (logs [here](https://mbta.splunkcloud.com/en-US/app/search/search?sid=1671561646.5577853)). The amount it exceeded the healthy threshold is roughly the same between dev-blue and dev-green (dev-green actually spiked higher, though not significantly).

Reply:
> However, it looks like that has been the trend lately on prod - wondering if it has more to do with the rating switchover + bigger files?

Yes, exactly - around the time of the new rating we tend to hit or slightly exceed the healthy threshold. I know a while back we bumped the threshold up, we should keep an eye on this to make sure we're not actually getting into territory where memory usage is getting close to the total available on the instance.

In any event as far as this PR is concerned those logs satisfy my curiosity. 👍